ddsreplayer/test/blackbox/mcap/McapFileReadTest.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// #define MCAP_IMPLEMENTATION
+
+#include <cpp_utils/testing/gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <cpp_utils/event/MultipleEventHandler.hpp>
+
+#include "dds/HelloWorldSubscriber.h"
+
+#include "tool/DdsReplayer.hpp"
+
+#include <iostream>
+#include <thread>
+#include <chrono>
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastrtps::rtps;
+using namespace eprosima::fastrtps;
+using namespace eprosima::ddsrecorder::replayer;
+
+namespace test {
+
+// Publisher
+
+const unsigned int DOMAIN = 0;
+
+std::string topic_name = "/dds/topic";
+std::string data_type_name = "HelloWorld";
+
+std::string send_message = "Hello World";
+unsigned int index = 6;
+unsigned int downsampling = 3;
+
+} // test
+
+enum class DataTypeKind
+{
+    HELLO_WORLD,
+};
+
+void create_subscriber_replayer(
+        DataToCheck& data)
+{
+    // Create Subscriber
+    HelloWorldSubscriber subscriber(
+        test::topic_name,
+        static_cast<uint32_t>(test::DOMAIN),
+        11,
```

Comment:
How this mcap file was created as well as its content should be well documented.

Reply:
Perhaps it is less effort to create an mcap file on the fly reusing code from recorder tests, just a suggestion.